packages/devtools_app/lib/src/screens/memory/memory_service.dart

```diff
@@ -65,80 +64,79 @@ class InboundReference extends Response {
   }
 
   static InboundReference parse(Map<String, dynamic> json) {
-    return json == null ? null : InboundReference._fromJson(json);
+    return InboundReference._fromJson(json);
   }
 
   dynamic parentField;
 
-  int parentListIndex;
+  int? parentListIndex;
 
-  int parentWordOffset;
+  int? parentWordOffset;
 
   bool get isFieldRef => parentField.runtimeType == FieldRef;
 
-  FieldRef get fieldRef => isFieldRef ? parentField as FieldRef : null;
+  FieldRef? get fieldRef => isFieldRef ? parentField as FieldRef? : null;
 
   bool get isClassRef => parentField.runtimeType == ClassRef;
 
-  ClassRef get classRef => isFieldRef ? parentField as ClassRef : null;
+  ClassRef? get classRef => isFieldRef ? parentField as ClassRef? : null;
 
   bool get isFuncRef => parentField.runtimeType == FuncRef;
 
-  FuncRef get funcRef => isFuncRef ? parentField as FuncRef : null;
+  FuncRef? get funcRef => isFuncRef ? parentField as FuncRef? : null;
 
   bool get isNullVal => parentField.runtimeType == NullVal;
 
   bool get isNullValRef => parentField.runtimeType == NullValRef;
 
-  NullVal get nullVal => isInstanceRef ? parentField as NullVal : null;
+  NullVal? get nullVal => isInstanceRef ? parentField as NullVal? : null;
 
   bool get isInstance => parentField.runtimeType == Instance;
 
-  Instance get instance => isInstance ? parentField as Instance : null;
+  Instance? get instance => isInstance ? parentField as Instance? : null;
 
   bool get isInstanceRef => parentField.runtimeType == InstanceRef;
 
-  InstanceRef get instanceRef =>
-      isInstanceRef ? parentField as InstanceRef : null;
+  InstanceRef? get instanceRef =>
+      isInstanceRef ? parentField as InstanceRef? : null;
 
   bool get isLibrary => parentField.runtimeType == Library;
 
   bool get isLibraryRef => parentField.runtimeType == LibraryRef;
 
-  Library get library => isLibrary ? parentField as Library : null;
+  Library? get library => isLibrary ? parentField as Library? : null;
 
   bool get isObj => parentField.runtimeType == Obj;
 
   bool get isObjRef => parentField.runtimeType == ObjRef;
 
-  Obj get obj => isObj ? parentField as Obj : null;
+  Obj? get obj => isObj ? parentField as Obj? : null;
 
   bool get isSentinel => parentField.runtimeType == Sentinel;
 }
 
 typedef BuildInboundEntry = void Function(
-  String referenceName,
+  String? referenceName,
   /* Field that owns reference to allocated memory */
-  String owningAllocator,
+  String? owningAllocator,
   /* Parent class that allocated memory. */
   bool owningAllocatorIsAbstract,
   /* is owning class abstract */
 );
 
-ClassHeapDetailStats _searchClass(
+ClassHeapDetailStats? _searchClass(
   List<ClassHeapDetailStats> allClasses,
-  String className,
+  String? className,
```

Comment:
should this be non-nullable?

Reply:
nope. It gets classRef.name, that is nullable.